db/migrate/20181218071253_add_etag_to_uploads.rb

```diff
@@ -0,0 +1,17 @@
+class AddEtagToUploads < ActiveRecord::Migration[5.2]
+  def change
+    add_column :uploads, :etag, :string
+    add_index :uploads, [:etag]
+
+    add_column :optimized_images, :etag, :string
+    add_index :optimized_images, [:etag]
+  end
+
+  def down
```

Comment:
We don't need `down` here since you're using `change`. Rails migration can automatically figure out how to migrate down.

Reply:
I removed it.